File: c/algo/selection-sort.c
#include <stdio.h>

void printArray(int arr[], int n) {
    for (int i = 0; i < n; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

void sort(int arr[], size_t len) {
    for (size_t i = 0; i < len; i++) {
        int min = i;
        for (size_t j = i+1; j < len; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        // swap the i and min
        int temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }
}

int main() {
    int arr[] = {2, 7, 23, 8, 268, 2, 97, 2};
    size_t n = sizeof(arr) / sizeof(arr[0]);
    
    printf("Original array: ");
    printArray(arr, n);
    
    sort(arr, n);
    
    printf("Sorted array: ");
    printArray(arr, n);
    
    return 0;
}

File: js/algo/selection-sort.js
function sort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let max = arr[i],
      maxIdx = i
    for (let j = i + 1; j < arr.length; j++) {
      if (max < arr[j]) {
        max = arr[j]
        maxIdx = j
      }
    }

    let temp = arr[i]
    arr[i] = max
    arr[maxIdx] = temp
  }
}

const array = [
  1, 7, 67, 83, 8, 0, 45, 345, 8, 9, 4, 46, 7970, 3, 4346, 56, 43, 2,
]
const result = sort(array)

console.log('Sorted Array: ', array)
